### DISCUSSION TITLE
Support angular with @nx/vite (configuration) plugin

### DISCUSSION THREAD

**Discussion by the-ult** [nrwl/nx#28952] [Feature Requests]
Thanks to the amazing work of Brandon Roberts and @yjaaidi on the vitest support for Angular within Nx, we can now generate new projects with Vitest.
- https://github.com/nrwl/nx/discussions/26339

However, when you wish to migrate an existing library, or add vitest to an existing Angular library with the `@nx/vite` plugin it does not generate the proper `vite.config.mts` file

#### Angular
Generated when creating a new library with testrunner=vitest:
`vite.config.mts`
```ts
/// <reference types='vitest' />
import angular from '@analogjs/vite-plugin-angular';
import { nxCopyAssetsPlugin } from '@nx/vite/plugins/nx-copy-assets.plugin';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineConfig } from 'vite';

export default defineConfig({
  root: __dirname,
  cacheDir: '../../../node_modules/.vite/libs/shared/foo',
  plugins: [angular(), nxViteTsPaths(), nxCopyAssetsPlugin(['*.md'])],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    watch: false,
    globals: true,
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    setupFiles: ['src/test-setup.ts'],
    reporters: ['default'],
    coverage: {
      reportsDirectory: '../../../coverage/libs/shared/foo',
      provider: 'v8',
    },
  },
});

```

And when using `@nx/vite`
`vitest.config.ts` -> not an `mts` file
```ts
/// <reference types='vitest' />
import { nxCopyAssetsPlugin } from '@nx/vite/plugins/nx-copy-assets.plugin';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineConfig } from 'vite';

export default defineConfig({
  root: __dirname,
  cacheDir: '../../../node_modules/.vite/libs/shared/utils',
  plugins: [nxViteTsPaths(), nxCopyAssetsPlugin(['*.md'])],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    watch: false,
    globals: true,
    environment: 'happy-dom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: {
      reportsDirectory: '../../../coverage/libs/shared/utils',
      provider: 'v8',
    },
  },
});

```

These are missing
- `plugins: [angular(),`
- `setupFiles: ['src/test-setup.ts'],`

Would be great if you could select the option, that it is an angular project ([uiFramework](https://nx.dev/nx-api/vite/generators/vitest#uiframework)), or even better if that is detected automatically

**Comment by yjaaidi** [comment]
Hey! Indeed, this is on my personal backlog 😉
I hope to be able to handle this next week.

**Comment by yjaaidi** [comment]
It would be nice if we could also add the vitest option to the preset (for users creating a new workspaces)

**Reply by the-ult** [reply]
Indeed.

And maybe even a schematic to migrate jest to vitest? 😇

**Reply by yjaaidi** [reply]
I have some plans for that 😉

**Comment by loxy** [comment]
Any news in this?

**Comment by atomicrobokid** [comment]
👋 Has there been any progress on NX Vitest support for Angular?

**Comment by adam-marshall** [comment]
I am surprised to read that Angular is still unsupported 

https://nx.dev/docs/technologies/build-tools/vite/generators#configuration

Why is this still the case?

**Comment by StavNoyAkur8** [comment]
Angular 21 officially adopted Vitest as the default test runner: 
[blog post](https://blog.angular.dev/announcing-angular-v21-57946c34f14b#c8d2)

The [guid for migration from Karma](https://angular.dev/guide/testing/migrating-to-vitest) includes automatic migration script, but it's experimental

**Comment by jase88** [comment]
Is it possible to use the new vitest implementation in Angular 21 with nx?
